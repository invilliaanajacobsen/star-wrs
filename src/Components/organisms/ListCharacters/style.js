import styled from 'styled-components';
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'

export const Line = styled(Row)`
  display: flex;
  align-items: stretch;
  justify-content: center;
  text-align: center;
  margin-bottom: 20px;
`


export const Column = styled(Col)`
  margin-bottom: 20px;
  transform:perspective(200px) rotateX(10deg);

  :hover {
    transform:perspective(0px) scale(1.2);
  }
`