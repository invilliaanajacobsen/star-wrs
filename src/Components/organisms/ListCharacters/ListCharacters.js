import React, { useState, Fragment } from "react";
import { useFetch } from "../../../services/Services";

import Loader from "../../atoms/Loader";
import Cardboard from "../../molecules/Cardboard";
import Pagination from "../../organisms/Pagination";

import { Line, Column } from "./style.js";

const ListCharacters = () => {
  const [url] = useState("https://swapi.co/api/people/");
  const [data, loading, setData, setLoading, resultDatas] = useFetch(
    url,
    "https://www.googleapis.com/customsearch/v1?q=tie&cx=006556936853971270487%3Atip6rjdt3xu&searchType=image&key=AIzaSyCV5v1NbhR02akfxiH1LpvOZWcczCMwNWY"
  );

  const nextPage = async number => {
    setLoading(true);
    const response = await fetch(`https://swapi.co/api/people/?page=${number}`);
    const json = await response.json();
    setData(json);
    setLoading(false);
  };

  return (
    <Fragment>
      {loading === false ? (
        <Fragment>
          <Line>
            {resultDatas.map((result, index) => {
              return (
                <Column
                  key={`${index}-${result.name}-col`}
                  xl="3"
                  lg="3"
                  md="12"
                  sm="12"
                  xs="12"
                >
                  <Cardboard
                    key={`${index}-${result.name}-card`}
                    title={result.name}
                    subtitle={result.birth_year}
                    img={result.img}
                    button="naves estelares"
                  />
                </Column>
              );
            })}
          </Line>
          <Pagination data={data.count} onClick={e => nextPage(e)} />
        </Fragment>
      ) : (
          <Loader />
        )}
    </Fragment>
  );
};
export default ListCharacters;
