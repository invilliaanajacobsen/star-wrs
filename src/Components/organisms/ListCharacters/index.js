import ListCharacters from './ListCharacters';

export default ListCharacters;
