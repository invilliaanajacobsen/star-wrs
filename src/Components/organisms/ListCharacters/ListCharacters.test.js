import React from "react";
import { shallow } from "enzyme";

import ListCharacters from "./ListCharacters";

describe("FixedIncome component", () => {
  let getListCharacters;
  let getImgsCharacters;

  beforeEach(() => {
    getListCharacters = jest.fn();
    getImgsCharacters = jest.fn();
  });

  it("should have an initial state", () => {
    const component = shallow(<ListCharacters />);
  });

  it("should match snapshot with loading", () => {
    expect(shallow(<ListCharacters />)).toMatchSnapshot();
  });
});
