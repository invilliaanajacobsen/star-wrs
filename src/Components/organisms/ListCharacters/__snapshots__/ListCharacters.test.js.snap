// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedIncome component should match snapshot with loading 1`] = `
<Fragment>
  <Loader />
</Fragment>
`;
